sklearn/datasets/descr/diabetes.rst

```diff
@@ -2,7 +2,37 @@ Diabetes dataset
 
 Notes
 -----
+
 Ten baseline variables, age, sex, body mass index, average blood
 pressure, and six blood serum measurements were obtained for each of n =
 442 diabetes patients, as well as the response of interest, a
 quantitative measure of disease progression one year after baseline.
+
+Note that the 10 x variables have been standardized to have mean 0 and squared length = 1 (sum(x^2)=1).
```

Comment:
Indeed it would be worth explaining this is more details in the documentation, for instance "each feature is centered has been scaled by standard deviation times `n_samples`" as this is not a very standard scale (pun intended).

Maybe we should introduce an option (in a separate PR) to load the [original diabetes data](http://www4.stat.ncsu.edu/~boos/var.select/diabetes.tab.txt) (without that strange scale nor the centering) to let the users use the real `StandardScaler` or any other feature engineering strategy if they which.

The default behavior should stay the same for backward compat.